bubblewrap.c

```diff
@@ -221,12 +221,17 @@ static void
 block_sigchild (void)
 {
   sigset_t mask;
+  int status;
 
   sigemptyset (&mask);
   sigaddset (&mask, SIGCHLD);
 
   if (sigprocmask (SIG_BLOCK, &mask, NULL) == -1)
     die_with_error ("sigprocmask");
+
+  /* Reap any outstanding zombies that we may have inherited */
+  while (waitpid (-1, &status, WNOHANG) > 0)
```

Comment:
Hmm, but don't we still need to call `waitpid()` after being notified by the signalfd?  (And actually aren't we subject to the default races that all current SIGCHLD usage has in that there may be multiple)?

I suppose in this case we know we didn't execute any other child processes, so it'd be pretty safe to do a waitpid loop if we get any sigchld.